newIDE/app/src/Profile/Subscription/SubscriptionChecker.js

```diff
@@ -0,0 +1,151 @@
+import {
+  sendSubscriptionCheckDialogShown,
+  sendSubscriptionCheckDismiss,
+} from '../../Utils/Analytics/EventSender';
+import { SubscriptionSuggestionContext } from './SubscriptionSuggestionContext';
+import Text from '../../UI/Text';
+
+export type SubscriptionCheckerInterface = {|
+  checkHasSubscription: () => boolean,
+|};
+
+type Props = {|
+  title: React.Node,
+  id:
+    | 'Disable GDevelop splash at startup'
+    | 'Debugger'
+    | 'Hot reloading'
+    | 'Preview over wifi',
+  onChangeSubscription?: () => Promise<void> | void,
+  mode: 'try' | 'mandatory',
+|};
+
+const styles = {
+  icon: { width: 40, height: 40, marginRight: 20 },
+  iconText: { flex: 1 },
+};
+
+const SubscriptionChecker = React.forwardRef<
+  Props,
+  SubscriptionCheckerInterface
+>(({ mode, id, title, onChangeSubscription }, ref) => {
+  const authenticatedUser = React.useContext(AuthenticatedUserContext);
+  const [dialogOpen, setDialogOpen] = React.useState<boolean>(false);
+  const { openSubscriptionDialog } = React.useContext(
+    SubscriptionSuggestionContext
+  );
+
+  const closeDialog = () => {
+    sendSubscriptionCheckDismiss();
+    setDialogOpen(false);
+  };
+
+  const checkHasSubscription = () => {
+    if (authenticatedUser.subscription) {
+      const hasPlan = !!authenticatedUser.subscription.planId;
+      if (hasPlan) {
+        setDialogOpen(false);
+        return true;
+      }
+    }
+
+    setDialogOpen(true);
+    sendSubscriptionCheckDialogShown({ mode, id });
+
+    return false;
+  };
+
+  React.useImperativeHandle(ref, () => ({ checkHasSubscription }));
+
+  return (
+    <Dialog
+      open={dialogOpen}
+      title={mode === 'try' ? <Trans>We need your support!</Trans> : title}
+      actions={[
+        <DialogPrimaryButton
+          label={<Trans>Get a subscription or login</Trans>}
+          key="subscribe"
+          primary
+          onClick={() => {
+            if (onChangeSubscription) onChangeSubscription();
+            setDialogOpen(false);
+            openSubscriptionDialog({
+              reason: id,
+              preStep: 'subscriptionChecker',
+            });
+          }}
+        />,
+      ]}
+      secondaryActions={[
+        <FlatButton
+          label={
+            mode === 'try' ? (
+              <Trans>Continue anyway</Trans>
+            ) : (
+              <Trans>Not now, thanks!</Trans>
+            )
+          }
+          key="close"
+          primary={false}
+          onClick={closeDialog}
+        />,
+      ]}
+      onRequestClose={closeDialog}
+    >
+      <Column noMargin>
+        <Line noMargin alignItems="center">
+          {mode === 'try' ? (
+            <Text>
+              <Trans>Please get a subscription to keep GDevelop running.</Trans>
+            </Text>
+          ) : (
+            <Text>
+              <Trans>
+                To use this feature, we ask you to get a subscription to
+                GDevelop.
```

Comment:
probably a wording from before, but I'd go for: 
`To use this feature, you need a GDevelop subscription.`